File: fss.erl
-module(fss).
-author('kleshney@gmail.com').
-export([start/0, listen/1]).
-define(PORTNO, 843).
-define(RESPONSE, "<cross-domain-policy><allow-access-from domain='*' to-ports='*' /></cross-domain-policy>").

start() -> spawn_link(?MODULE, listen, [?PORTNO]).

listen(Port) ->
    case gen_tcp:listen(Port, [binary, {packet, 0}, {active, false}]) of
        {ok, ServerSocket} -> 
            loop(ServerSocket);
        {error, Reason} -> 
            throw({'cant bind socket', Reason})
    end.

loop(ServerSocket) ->
    case gen_tcp:accept(ServerSocket) of
        {ok, Socket} ->
            gen_tcp:send(Socket, [?RESPONSE] ++ [13, 10]),
            gen_tcp:close(Socket),
            loop(ServerSocket);
        _ -> 
            loop(ServerSocket)
    end.

File: fss.go
package main

import (
	"fmt";
	"net";
	"os";
)

func main() {
	var (
		host = "0.0.0.0";
		port = "843";
		remote = host + ":" + port;
		response = []byte("<cross-domain-policy><allow-access-from domain='*' to-ports='*' /></cross-domain-policy>\r\n");
	)
	fmt.Println("Initiating server... (Ctrl-C to stop)");

	lis, error := net.Listen("tcp", remote);
	defer lis.Close();
	if error != nil { 
		fmt.Printf("Error creating listener: %s\n", error ); 
		os.Exit(1); 
	}
	for {
		
		con, error := lis.Accept();
		if error != nil { fmt.Printf("Error: Accepting data: %s\n", error); os.Exit(2); }
		fmt.Printf("Handled request from: %s \n", con.RemoteAddr()); 
		//con.Write(strings.Bytes(response));
		con.Write(response);	
		con.Close();
	}
}
